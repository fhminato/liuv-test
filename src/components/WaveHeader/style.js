import styled from 'styled-components/native';

export const Container = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
  background: #5000ca;
  height: 100px;
`;

export const TextContainer = styled.View`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Text = styled.Text`
  color: #fff;
  text-transform: uppercase;
  font-family: 'Antipasto Pro Demibold';
  font-size: 30px;
`;
