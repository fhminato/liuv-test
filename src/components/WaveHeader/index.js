import React from 'react';
import {Container, TextContainer, Text} from './style';

import Svg, {Path} from 'react-native-svg';

export default function WaveHeader({title}) {
  return (
    <Container>
      <TextContainer>
        <Text testID={'title'}>{title}</Text>
      </TextContainer>
      <Svg
        height="100%"
        width="100%"
        viewBox="0 0 1440 320"
        style={{position: 'absolute', top: 80}}>
        <Path
          fill="#5000ca"
          d="M0,96L48,112C96,128,192,160,288,186.7C384,213,480,235,576,213.3C672,192,768,128,864,128C960,128,1056,192,1152,208C1248,224,1344,192,1392,176L1440,160L1440,0L1392,0C1344,0,1248,0,1152,0C1056,0,960,0,864,0C768,0,672,0,576,0C480,0,384,0,288,0C192,0,96,0,48,0L0,0Z"
        />
      </Svg>
    </Container>
  );
}
