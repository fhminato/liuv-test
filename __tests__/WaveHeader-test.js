import React from 'react';
import WaveHeader from '../src/components/WaveHeader';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<WaveHeader title={'Repositorios'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('test text', () => {
  const tree = renderer.create(<WaveHeader title={'Repositorios'} />);

  const title = tree.root.props['title'];
  expect(title).toBe('Repositorios');
});
