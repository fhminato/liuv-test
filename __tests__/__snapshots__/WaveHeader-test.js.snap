// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#5000ca",
        "display": "flex",
        "height": 100,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Antipasto Pro Demibold",
            "fontSize": 30,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="title"
    >
      Repositorios
    </Text>
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    focusable={false}
    height="100%"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "position": "absolute",
          "top": 80,
        },
        Object {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    vbHeight={320}
    vbWidth={1440}
    width="100%"
  >
    <RNSVGGroup>
      <RNSVGPath
        d="M0,96L48,112C96,128,192,160,288,186.7C384,213,480,235,576,213.3C672,192,768,128,864,128C960,128,1056,192,1152,208C1248,224,1344,192,1392,176L1440,160L1440,0L1392,0C1344,0,1248,0,1152,0C1056,0,960,0,864,0C768,0,672,0,576,0C480,0,384,0,288,0C192,0,96,0,48,0L0,0Z"
        fill={4283433162}
        propList={
          Array [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
